src/sentry/options/manager.py

```diff
@@ -511,3 +513,21 @@ def can_update(self, key: str, value, channel: UpdateChannel) -> NotWritableReas
             return NotWritableReason.DRIFTED
 
         return None
+
+    def bulk_retrieve_options(self, keys: list[str], flag: int | None = None):
+        key_set = set(keys)
+        keys = []
+
+        for key, value in self.registry.items():
+            flag_matches = flag is None or value.flags & flag
+            key_in_set = key in key_set
+
+            if not key_in_set or not flag_matches:
+                continue
```

Comment:
Should we raise an error here? Silently not fetching an option because the flags are wrong could result in KeyErrors in the caller as they expect to get the options they requested.

Reply:
I don't think we should. In flagpole we want to skip any flags that aren't defined, so in a bulk flag lookup scenario we'll end up raising an exception and evaluating all flags to None which doesn't seem correct. Alternatively, we could just make flagpole individually query options since we're doing that here anyway, avoiding this bulk behavior entirely.

Reply:
>  Alternatively, we could just make flagpole individually query options since we're doing that here anyway,

I think that might be the simpler solution. If we prefix the options that flagpole uses we don't need to expand the option flags which are already complicated and have some poorly defined intersections already.

Reply:
`configoptions` uses `filter`, which does something very similar (return a list of options filtered on a flag). 
Should we combine the two?

Reply:
Agreed, closing this change.